compiler/rustc_ast_lowering/src/expr.rs

```diff
@@ -43,7 +43,10 @@ impl<'hir> LoweringContext<'_, 'hir> {
                 ExprKind::Call(ref f, ref args) => {
                     if e.attrs.get(0).map_or(false, |a| a.has_name(sym::rustc_box)) {
                         if let [inner] = &args[..] {
-                            hir::ExprKind::Box(self.lower_expr(&inner))
+                            let kind = hir::ExprKind::Box(self.lower_expr(&inner));
+                            let hir_id = self.lower_node_id(e.id);
+                            self.lower_attrs(hir_id, &e.attrs[1..]);
```

Comment:
So this assumes that `rustc_box` is the first attribute? If `rustc_box` is assumed to be the only attribute, then we could just skip lowering attributes at all.

Reply:
My reasoning for assuming it was first was that `rustc_box` shouldn't appear so often and I wouldn't want to do a possibly expensive search over all attributes. Once we have `rustc_box`, it should be comparatively inexpensive to take care of the other attributes.

Reply:
(When making that decision, I thought about doc attributes but function calls obviously don't have doc attributes so I was a bit misguided)